### PR TITLE
statistics: enable the auto-analysis priority queue

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50132

Problem Summary:

### What changed and how does it work?

I have enabled the auto-analysis priority queue by default. Also from this PR, the priority queue starts respecting the auto-analyze time window.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] [Manual test](https://github.com/pingcap/tidb/pull/51537#issuecomment-1982560156)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
Tested locally:
1. start the TiDB cluster
2. check the TiDB version
```sql
mysql> select tidb_version();
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                       |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v8.0.0-alpha-424-gd18713949c
Edition: Community
Git Commit Hash: d18713949ccbf81ab1b38efa48ea1ca9ee6c98d5
Git Branch: rustin-patch-enable-pq
UTC Build Time: 2024-03-07 06:19:15
GoVersion: go1.21.7
Race Enabled: false
Check Table Before Drop: false
Store: tikv |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
3. Run the below python script
```python
import pymysql
import random

# Connect to TiDB
conn = pymysql.connect(host='localhost', port=4000, user='root', passwd='', db='test')
cursor = conn.cursor()

# Create partitioned table
# cursor.execute("""
#     CREATE TABLE users (
#       id INT NOT NULL,
#       name VARCHAR(50) NOT NULL,
#       age INT NOT NULL,
#       country VARCHAR(2) NOT NULL
#     )
#     PARTITION BY RANGE (age) (
#       PARTITION p0 VALUES LESS THAN (20),
#       PARTITION p1 VALUES LESS THAN (40),
#       PARTITION p2 VALUES LESS THAN (60)
#     );
# """)

# Generate and insert data
for i in range(2000):
  id = i
  name = 'User {}'.format(i)
  age = random.randint(15, 55)
  country = random.choice(['US', 'CN', 'IN'])

  cursor.execute("INSERT INTO users VALUES (%s, %s, %s, %s)", (id, name, age, country))

conn.commit()

print("Partitioned table created and data inserted successfully!")

conn.close()
```

4. Check logs:
```log
[2024/03/07 14:27:58.017 +08:00] [INFO] [refresher.go:101] ["Auto analyze triggered"] [category=stats] [job="DynamicPartitionedTableAnalysisJob:\n\tAnalyzeType: analyzeDynamicPartition\n\tPartitions: p1, p2\n\tPartitionIndexes: map[]\n\tSchema: test\n\tGlobal Table: users\n\tGlobal TableID: 104\n\tTableStatsVer: 2\n\tChangePercentage: 1.00\n\tTableSize: 14052.00\n\tLastAnalysisDuration: 30m0s\n\tWeight: 1.3791\n"]
[2024/03/07 14:27:58.049 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=106] [curModifyCnt=3898] [results.BaseModifyCnt=3898] [modifyCount=0]
[2024/03/07 14:27:58.049 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=106] [results.Count=3898] [count=3898]
[2024/03/07 14:27:58.169 +08:00] [INFO] [analyze.go:745] ["analyze table `test`.`users` has finished"] [partition=p1] ["job info"="auto analyze table all columns with 256 buckets, 500 topn, 1 samplerate"] ["start time"=2024/03/07 14:27:58.027 +08:00] ["end time"=2024/03/07 14:27:58.168 +08:00] [cost=140.721792ms] ["sample rate reason"="use min(1, 110000/3898) as the sample-rate=1"]
[2024/03/07 14:27:58.170 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=107] [curModifyCnt=3128] [results.BaseModifyCnt=3128] [modifyCount=0]
[2024/03/07 14:27:58.170 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=107] [results.Count=3128] [count=3128]
[2024/03/07 14:27:58.264 +08:00] [INFO] [analyze.go:745] ["analyze table `test`.`users` has finished"] [partition=p2] ["job info"="auto analyze table all columns with 256 buckets, 500 topn, 1 samplerate"] ["start time"=2024/03/07 14:27:58.048 +08:00] ["end time"=2024/03/07 14:27:58.262 +08:00] [cost=214.074167ms] ["sample rate reason"="use min(1, 110000/3128) as the sample-rate=1"]
```

**Review by AilinKid — APPROVED** [review]

File: pkg/statistics/handle/autoanalyze/refresher/refresher.go [hunk 1]
```diff
@@ -125,6 +132,27 @@ func (r *Refresher) RebuildTableAnalysisJobQueue() error {
...
+				)
+				return err
+			}
+			// We will check it again when we try to execute the job.
+			// So store the time window for later use.
+			r.autoAnalysisTimeWindow = autoAnalysisTimeWindow{
+				start: start,
+				end:   end,
+			}
+			if !r.autoAnalysisTimeWindow.isWithinTimeWindow(time.Now()) {
```
Comment (AilinKid) [inline pkg/statistics/handle/autoanalyze/refresher/refresher.go:153]: any mock test about the window check

Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/refresher/refresher.go:153]: https://github.com/pingcap/tidb/blob/d18713949ccbf81ab1b38efa48ea1ca9ee6c98d5/pkg/statistics/handle/autoanalyze/autoanalyze_test.go#L357

**Pull request merged**